SimCalorimetry/EcalSimAlgos/src/EcalLiteDTUCoder.cc

```diff
@@ -0,0 +1,241 @@
+                                    const EcalSamples& clf,
+                                    EcalDataFrame_Ph2& df) const {
+  df.setSize(clf.size());
+  encode(clf, df, engine);
+}
+
+void EcalLiteDTUCoder::encode(const EcalSamples& ecalSamples,
+                           EcalDataFrame_Ph2& df,
+                           CLHEP::HepRandomEngine* engine) const {
+  assert(nullptr != m_peds);
+
+  const unsigned int csize(ecalSamples.size());
+
+  DetId detId = ecalSamples.id();
+  double Emax = fullScaleEnergy(detId);
+
+  //....initialisation
+  if (ecalSamples[5] > 0.)
+    LogDebug("EcalLiteDTUCoder") << "Input caloSample"
+                              << "\n"
+                              << ecalSamples;
+
+  //N Gains set to 2 in the .h
+  double pedestals[ecalPh2::NGAINS];
+  double widths[ecalPh2::NGAINS];
+  //float  gains[ecalPh2::NGAINS];
+  double LSB[ecalPh2::NGAINS];
+  double trueRMS[ecalPh2::NGAINS];
+
+  double icalconst = 1.;
+  findIntercalibConstant(detId, icalconst);
+
+  for (unsigned int igain(0); igain < NGAINS; ++igain) {
+    // fill in the pedestal and width
+
+    findPedestal(detId, igain, pedestals[igain], widths[igain]);
+    //I insert an absolute value in the trueRMS
+    trueRMS[igain] = std::sqrt(std::fabs(widths[igain] * widths[igain] - 1. / 12.));
+
+    // set nominal value first
+    //findGains( detId , gains  );
+
+    LSB[igain] = Emax / (MAXADC * ecalPh2::gains[igain]);
+  }
+
+  CaloSamples noiseframe[] = {
+      CaloSamples(detId, csize),
+      CaloSamples(detId, csize),
+  };
+
+  const Noisifier* noisy[NGAINS] = {m_ebCorrNoise[0], m_ebCorrNoise[1]};
+
+  if (m_addNoise) {
+    //#warning noise generation to be checked
+    noisy[0]->noisify(noiseframe[0], engine);  // high gain
+    //if( nullptr == noisy[1] ) noisy[0]->noisify( noiseframe[1] ,
+    //                                             engine,
+    //                                              &noisy[0]->vecgau() ) ; // lwo gain
+    noisy[1]->noisify(noiseframe[1], engine);  // low gain
+  }
+
+  bool isSaturated[] = {false, false};
+  std::vector<std::vector<int>> adctrace(csize);
+  unsigned int saturatedSample[] = {0, 0};
+
+  for (unsigned int i(0); i != csize; ++i)
+    adctrace[i].resize(NGAINS);
+
+  for (unsigned int igain = 0; igain < NGAINS; ++igain) {
+    for (unsigned int i(0); i != csize; ++i) {
+      adctrace[i][igain] = -1;
+    }
+  }
+
+  // fill ADC trace in gain 0 (x10) and gain 1 (x1)
+  //NOTE: Before was  pedestals[igain], widths[igain]
+
+  for (unsigned int igain = 0; igain < NGAINS; ++igain) {
+    for (unsigned int i(0); i != csize; ++i) {
+      double asignal = 0;
+
+      if (!m_PreMix1) {
+        asignal =
+            pedestals[igain] + ecalSamples[i] / (LSB[igain] * icalconst) +
+            trueRMS[igain] *
+                noiseframe
+                    [igain]
+                    [i];  //Analog signal value for each sample in ADC. It is corrected by the intercalibration constants
```

Comment:
put comment on separate line to avoid awkward line break from code format

Reply:
done